termwiz/src/escape/parser/mod.rs

```diff
@@ -55,7 +56,13 @@ impl Parser {
             callback: &mut callback,
             state: &mut self.state.borrow_mut(),
         };
-        self.state_machine.parse(bytes, &mut perform);
+
+        if perform.state.tmux_cc {
+            self.state_machine.force_dcs_passthrough(bytes, &mut perform);
```

Comment:
I haven't looked to make sure this suggestion is good, but I think that we could add `DeviceControlMode::Bytes(Vec<u8>)` and pass that bytes through at the termwiz layer rather than having this special state cut across to the vtparse layer.

For that to work, I think we need a bit more state for tmux in here, so that we can match for a sequence like `\n%exit` to break out of tmux-cc mode.

When in tmux-cc mode, we can simply bypass the vtparse parser and redirect straight through to `DeviceControlMode::Bytes` which would get handled in the same way as `DeviceControlMode::Data`, but with the advantage that it's a bit more efficient to pass the vec through there, rather than the individual `Data` items.